### DISCUSSION TITLE
Do vite plugins work on HTML?

### DISCUSSION THREAD

**Discussion by smac89** [withastro/docs#9032] [Ideas]
Astro already compresses html, but I was hoping to make use of the vite integration for further compression. I tried adding [this plugin](https://jsr.io/@zade/vite-plugin-strip-html-comments) to the vite.plugins config, but I noticed that the final HTML still contains comments.

Do vite plugins which modify HTML work in Astro? How do I strip comments from the final HTML?

**Comment by smac89** [comment]
Seems related to https://github.com/withastro/astro/issues/2039

**Comment by bnjmnrsh** [comment]
I'm unsure if your issue is related to the bug you've cited or not, but I was also trying to [affect my build the other day by passing the `cspNonce` flag to Vite](https://github.com/withastro/astro/issues/11669), and was also frustrated as it wasn't being picked up.  

If your issue is _not_ related to the bug, perhaps this conversation could be expanded into one about better documentation for the Vite options. A few people seem to be spinning their wheels because they don't have an adequate mental model of what can be passed to Vite and what can't. IMHO, better documents here could help save some human cycles.

https://docs.astro.build/en/reference/configuration-reference/#vite

**Reply by Fryuni** [reply]
> A few people seem to be spinning their wheels because they don't have an adequate mental model of what can be passed to Vite and what can't.

The confusion is not what you can or can't pass to Vite. You can change any Vite option you might want, but not everything uses Vite. Astro is a combo of multiple tools plus all of its own logic.

**Comment by Fryuni** [comment]
> Do vite plugins which modify HTML work in Astro? How do I strip comments from the final HTML?

Astro doesn't use Vite to emit the HTML, neither during build nor when server-rendering, so plugins targeting Vite's HTML output won't affect Astro.

You can affect the Astro-generated HTML from an Astro middleware.

> How do I strip comments from the final HTML?

Be aware that doing so will break Astro functionality and possibly other libraries you might have. Many tools use HTML comments as markers, which is why most HTML minifiers won't remove them by default.

**Reply by olets** [reply]
> You can affect the Astro-generated HTML from an Astro middleware.

Thanks! That helped me find the example in the docs

https://docs.astro.build/en/guides/middleware/#example-redacting-sensitive-information
